lilypad/server/services/versions.py

```diff
@@ -88,19 +87,34 @@ def find_prompt_active_version(
     def change_active_version(
         self, project_id: int, new_active_version: VersionTable
     ) -> VersionTable:
-        """Change the active version"""
-        with suppress(HTTPException):
-            active_version = self.find_prompt_active_version(
-                project_id, new_active_version.function_name
-            )
-            if active_version.id == new_active_version.id:
-                return active_version
-            active_version.is_active = False
-            self.session.add(active_version)
+        """Change the active version for a function, deactivating any currently active versions.
+
+        Args:
+            project_id: The project ID
+            new_active_version: The version to make active
+
+        Returns:
+            The newly activated version
+        """
+        # Deactivate all currently active versions for the same function
+        stmt = select(VersionTable).where(
+            VersionTable.project_id == project_id,
+            VersionTable.function_name == new_active_version.function_name,
+            VersionTable.is_active,
+        )
+        current_active_versions = self.session.exec(stmt).all()
+
+        for version in current_active_versions:
+            version.is_active = False
+            self.session.add(version)
+
+        # Activate the new version
         new_active_version.is_active = True
         self.session.add(new_active_version)
-        self.session.flush()
```

Comment:
@brenkao i remember using `flush` for a reason here but it's eluding me

Reply:
flush is if you want to grab the updated id, we already commit everything at the end so committing is redundant

Reply:
or if there is an exception in the api, then we don't commit the transaction. if we call commit here, then we will always save, even if there are errors elsewhere

Reply:
ah ok yes this is what I was thinking of thank you

@koxudaxi please take this into account with this update so that we properly handle potential errors with flush instead of commit

Reply:
we should be able to keep flush, but add refresh and that should be good

Reply:
Thank you for the commit.
I reverted `commit` to `flush()`